### PR TITLE
*: controller-tools bump to v0.2.2+ to include bugfix for crd generation

### LINKED ISSUE #1932: Generated CRD subresources and schema not honored
## Bug Report

**What did you do?**
1. Build operator-sdk from master
2. Create new go project
3. Create new api and controller
4. Update _types.go file to include a new field in the status
5. Update controller to set the new status field on the CR and make a call to update the status (returning any error)
6. Run `operator-sdk generate k8s && operator-sdk generate openapi`
5. Apply CRD and CR
6. `operator-sdk up local`

**What did you expect to see?**
1. Everything running normally and the status field updated on the CR
2. `kubectl get crd apps.example.com -o yaml` to include the `subresource` setting somewhere

**What did you see instead? Under which circumstances?**
1. Failures trying to update the status, with an error indicating that the subresource does not exist
2. The in-cluster CRD does not include any mention of a `subresources.status` setting or the `schema` section that ARE present in disk, under the `v1` version.

**Environment**
* operator-sdk version:

```
operator-sdk version: "v0.10.0-48-g5f663690", commit: "5f663690a3bb97bf7b9e9b8a15a2f8d6bc481808", go version: "go1.13 darwin/amd64"
```

* go version:

```
go version go1.12.7 darwin/amd64
```

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:13:49Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:05:16Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: minikube (local)

* Are you writing your operator in ansible, helm, or go? go

**Possible Solution**
* Make `subresources` a top-level field
* Move `schema` to the top-level `validation` field.

**Additional context**
* I repeated similar steps using kubebuilder and noticed that it produces a CRD with a top-level `subresources` section.
* I repeated the same steps using a Kubernetes 1.15 cluster and got CRD validation errors trying to create the CRD:
   ```
   The CustomResourceDefinition "apps.example.org" is invalid:
   * spec.versions: Invalid value: []apiextensions.CustomResourceDefinitionVersion{apiextensions.CustomResourceDefinitionVersion{Name:"v1", Served:true, Storage:true, Schema:(*apiextensions.CustomResourceValidation)(0xc00ad95438), Subresources:(*apiextensions.CustomResourceSubresources)(0xc004f98ed0), AdditionalPrinterColumns:[]apiextensions.CustomResourceColumnDefinition(nil)}}: per-version schemas may not all be set to identical values (top-level validation should be used instead)
   * spec.versions: Invalid value: []apiextensions.CustomResourceDefinitionVersion{apiextensions.CustomResourceDefinitionVersion{Name:"v1", Served:true, Storage:true, Schema:(*apiextensions.CustomResourceValidation)(0xc00ad95438), Subresources:(*apiextensions.CustomResourceSubresources)(0xc004f98ed0), AdditionalPrinterColumns:[]apiextensions.CustomResourceColumnDefinition(nil)}}: per-version subresources may not all be set to identical values (top-level subresources should be used instead)
   ```

/cc @estroz @hasbro17

### PR DESCRIPTION
**Pull request by hasbro17**
**Description of the change:**
Pin controller-tools to commit [`6ed4ff330711554efe27e43985d36b95ea61782b`](https://github.com/kubernetes-sigs/controller-tools/tree/6ed4ff330711554efe27e43985d36b95ea61782b) which includes the bugfix https://github.com/kubernetes-sigs/controller-tools/pull/322 that correctly propagates version specific subfields(`subresources`, `schema`, and `additionalPrinterColumns`) to the top level `spec` for CRD generation.

**Motivation for the change:**
Fixes #1932

### CODE REVIEW

**Comment by hasbro17** [comment]
Since this PR changes how CRD manifests are generated and pushes up the per-version fields to the top level spec if they're identical, this introduces a bug in our CI since we're using a k8s 1.11 cluster on travis.
https://travis-ci.org/operator-framework/operator-sdk/jobs/589203495#L3327
```
Failed to create resource(s) in global resource manifest: CustomResourceDefinition.apiextensions.k8s.io \"memcacheds.cache.example.com\" is invalid:
...
must only have \"properties\", \"required\" or \"description\" at the root if the status subresource is enabled
```
https://github.com/kubernetes/kubernetes/issues/65293


The [bugfix](https://github.com/kubernetes/kubernetes/pull/65357) for the above never got ported to k8s 1.11 so it shows up for our go e2e running on k8s 1.11 in travis.


So this PR has a temporary workaround in the scaffolded test data to remove the `type: object` field from the CRD so that it works on a 1.11 cluster.
This is just until we can upgrade our outdated v1.11 cluster on travis or remove it altogether.

File: hack/tests/scaffolding/scaffold-memcached.go [hunk 1]
```diff
@@ -191,6 +203,36 @@ func main() {
 	}
 }
 
+// TODO(haseeb): Remove this when this test no longer runs on a k8s v1.11 cluster in CI
+// removeTypeFromCRDValidation will edit the memcached CRD manifest to remove
+// the "type: object" field from root of the CRD validation block.
+// This serves as a workaround for the following bug on a k8s 1.11 cluster:
+// https://github.com/kubernetes/kubernetes/issues/65293
+func removeTypeFromCRDValidation() {
+	crdPath := "deploy/crds/cache.example.com_memcacheds_crd.yaml"
```
Comment (estroz) [inline hack/tests/scaffolding/scaffold-memcached.go]: I'm fine with hardcoding this because this is a temporary workaround.

**Comment by hasbro17** [comment]
test e2e-aws-ansible

**Comment by hasbro17** [comment]
I've added the changes from #1959 here as well since the `e2e-aws-ansible` test is failing again due to the permissions error that comes about when the USER in the base image isn't added to /etc/passwd:
```
+ kubectl logs deployment/memcached-operator -c ansible
Setting up watches.  Beware: since -r was given, this may take a while!
Watches established.
/tmp/ansible-operator/runner/ansible.example.com/v1alpha1/Memcached/default/example-memcached/artifacts/3916589616287113937//stdout
[0;34mansible-playbook 2.8.0[0m
[0;34m  config file = /etc/ansible/ansible.cfg[0m
[0;34m  configured module search path = [u'/usr/share/ansible/openshift'][0m
[0;34m  ansible python module location = /usr/lib/python2.7/site-packages/ansible[0m
[0;34m  executable location = /usr/bin/ansible-playbook[0m
[0;34m  python version = 2.7.5 (default, Jun 11 2019, 14:33:56) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)][0m
[0;34mUsing /etc/ansible/ansible.cfg as config file[0m

[0;34mUsing /etc/ansible/ansible.cfg as config file[0m

PLAYBOOK: 225a75c8b66d46a0bde5b592a8ecb599 *************************************
[0;34m1 plays in /tmp/ansible-operator/runner/ansible.example.com/v1alpha1/Memcached/default/example-memcached/project/225a75c8b66d46a0bde5b592a8ecb599[0m

PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
[1;31mfatal: [localhost]: UNREACHABLE! => {"changed": false, "msg": "Authentication or permission failure. In some cases, you may have been able to authenticate and did not have permissions on the target directory. Consider changing the remote tmp path in ansible.cfg to a path rooted in \"/tmp\". Failed command was: ( umask 77 && mkdir -p \"` echo /.ansible/tmp/ansible-tmp-1569100069.49-140240887647478 `\" && echo ansible-tmp-1569100069.49-140240887647478=\"` echo /.ansible/tmp/ansible-tmp-1569100069.49-140240887647478 `\" ), exited with result 1", "unreachable": true}[0m

PLAY RECAP *********************************************************************
[0;31mlocalhost[0m                  : ok=0    changed=0    [1;31munreachable=1   [0m failed=0    skipped=0    rescued=0    ignored=0   

+ exit 1
```

Even though CRIO be setting the /etc/passwd entry based on the `$HOME` and `$USER_NAME` variables(https://github.com/cri-o/cri-o/pull/2563) we're seeing this flake a lot recently.

So we're reverting back to our manual setup scripts.

**Pull request merged**